src/extension.c

```diff
@@ -149,7 +135,13 @@ extension_is_loaded(void)
 	 */
 	id = get_extension_oid(EXTENSION_NAME, true);
 
-	if (OidIsValid(id))
+	/*
+	 * the proxy table check is necessary because during an extension drop the
+	 * proxy table will be deleted before the extension itself. You don't want
+	 * to mark the extension as CREATED in this case
+	 */
+
+	if (OidIsValid(id) && proxy_table_exists())
```

Comment:
So, I vaguely recall that the reason for returning true while the extension was being dropped was that otherwise it would not clean up (invalidate caches) because in `inval_cache_callback()` there is an `extension_is_loaded()` check. Is this no longer the case? I mean, like I expressed above, this might work incidentally due to ordering of events, and the order of `extension_is_loaded()` checks in the code. I am worried that this will break as soon as we add another `extension_is_loaded()` check in the code or the order of events changes.

I guess part of the problem is that `extension_is_loaded()` both checks the state and sets the state, so the state might change between successive calls.